homeassistant/components/sleepiq/switch.py

```diff
@@ -39,15 +39,17 @@ def __init__(
         self._attr_name = f"SleepNumber {bed.name} Pause Mode"
         self._attr_unique_id = f"{bed.id}-pause-mode"
 
-    @property
-    def is_on(self) -> bool:
-        """Return whether the switch is on or off."""
-        return bool(self.bed.paused)
-
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn on switch."""
         await self.bed.set_pause_mode(True)
+        self._async_update_attrs()
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn off switch."""
         await self.bed.set_pause_mode(False)
+        self._async_update_attrs()
```

Comment:
Did you intent to call `self.async_write_ha_state()` as well here to update the state machine?